.github/workflows/static_code_analysis.yml

```diff
@@ -0,0 +1,33 @@
+name: Static Code Analysis
+
+on:
+  release:
+    types: [published]
+  workflow_dispatch:
+
+jobs:
+  code-analysis:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.10'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install bandit==1.7.7
+    - name: Save code analysis
+      run: bandit -r . -x ./tests -f txt -o static_code_analysis.txt --exit-zero
+    - name: Create pull request
+      id: cpr
+      uses: peter-evans/create-pull-request@v4
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Can we switch to `GH_ACCESS_TOKEN`? I know in SDV we used this one, but the built in `GITHUB_TOKEN` won't trigger the tests when the PR is opened

Reply:
Alright, done in [0fb5971](https://github.com/sdv-dev/CTGAN/pull/356/commits/0fb5971cdfa6410a50c8cb95df39adac22fb16bb)